test/Discord.Net.Tests/Tests.DiscordWebhookClient.cs

```diff
@@ -0,0 +1,62 @@
+using Discord.Webhook;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace Discord
+{
+    /// <summary>
+    ///     Tests that the <see cref=""/>
+    /// </summary>
+    public class DiscordWebhookClientTests
+    {
+        [Theory]
+        [InlineData("https://discordapp.com/api/webhooks/123412347732897802/_abcde123456789-ABCDEFGHIJKLMNOP12345678-abcdefghijklmnopABCDEFGHIJK")]
+        // ptb, canary, etc will have slightly different urls
+        [InlineData("https://ptb.discordapp.com/api/webhooks/123412347732897802/_abcde123456789-ABCDEFGHIJKLMNOP12345678-abcdefghijklmnopABCDEFGHIJK")]
+        [InlineData("https://canary.discordapp.com/api/webhooks/123412347732897802/_abcde123456789-ABCDEFGHIJKLMNOP12345678-abcdefghijklmnopABCDEFGHIJK")]
+        // don't care about https
+        [InlineData("http://canary.discordapp.com/api/webhooks/123412347732897802/_abcde123456789-ABCDEFGHIJKLMNOP12345678-abcdefghijklmnopABCDEFGHIJK")]
+        // this is the minimum that the regex cares about
+        [InlineData("p.com/api/webhooks/123412347732897802/_abcde123456789-ABCDEFGHIJKLMNOP12345678-abcdefghijklmnopABCDEFGHIJK")]
+        public void TestWebhook_Valid(string webhookurl)
+        {
+            try
+            {
+                _ = new DiscordWebhookClient(webhookurl);
```

Comment:
These tests should do the same format check that the client's constructor does, without using the constructor.

The Webhook client is a special snowflake and actually makes an API request in its ctor currently, so this is technically an integration test, not a unit test.

Reply:
Maybe move the parsing logic into some static internal function (since tests have internal access)?

Reply:
Sounds good to me

Reply:
I really wish I could have used a ValueTuple here https://github.com/discord-net/Discord.Net/pull/1260/files#diff-062b7bba813fb6aac2adf2a6f980eb7fR120